thundra/reporter.py

```diff
@@ -102,11 +102,13 @@ def prepare_report_json(self, reports):
             batched_reports.append(json_string)
         return batched_reports
 
-    def prepare_composite_report_json(self, reports):
+    def prepare_composite_report_json(self, reports, **opts):
         invocation_report = None
         for report in reports:
             if report["type"] == "Invocation":
                 invocation_report = report
+            elif opts.get("test_run_event", False):
```

Comment:
I think, instead of this hacky solution, you might pass a boolean value (`disable_composite`) to disable composite monitoring data and use `prepare_report_json` instead. So this flag `disable_composite`) can be set to `true` while reporting test-run events. So there won't be needed to inject application info into test-run events.

Reply:
done.